### ISSUE TITLE
Switch to AndroidX

### ISSUE THREAD

**Issue by sarvajeet294** [greenrobot/greenDAO#920]
Greendao is not updated for androidx support libraries. Annotations like @UiThread @CallSuper are taken from annotation library but which comes from appcompat.

In androidx appcompat library and annotation librarie's package names are changed. 
Greendao should keep this scenario in mind and change it accordingly. Please make those changes.

**Comment by greenrobot-team** [comment]
Please refer to https://developer.android.com/jetpack/androidx/migrate on how to use AndroidX with libraries that do not yet use the new imports.

Keeping this open if we ever want to do a release supporting AndroidX.
-ut

**Comment by softboy99** [comment]
Hi,
I'm also waiting for this feature.Thanks.

**Comment by prgpascal** [comment]
Hello, 
this feature is really important for us because we're generating the entities with the greendao-generator. After every DAO/entities generation, we must change the import of NonNull notations, from `android.support.annotation.NonNull` to `androidx.annotation.NonNull`. 
Any workaround for this?

Thanks

**Comment by greenrobot-team** [comment]
@prgpascal As far as I know and based on a quick check, greenDAO does not use the Android support annotations. It uses its own `org.greenrobot.greendao.annotation.NotNull`. Maybe you add those in your generator code?

**Comment by prgpascal** [comment]
@greenrobot-team you're right, it was my fault :-/  We currently inject the annotation dependency into our generated entities through our generator class. Changing that dependency solved my issue. Thanks!

**Comment by greenrobot-team** [comment]
Closing this then.

**Comment by greenrobot-team** [comment]
Whoops, this issue is about something else. Re-opening.

**Comment by ZacSweers** [comment]
It's well past the point of holding out I think, as this also prevents other libraries and open source projects that depend on greenDAO from updating. Jetifier is intended to be a temporary cover while libraries migrate, not a permanent reason for them to not migrate.
